### PR TITLE
Build Directory Cleaning Functionality

### PR DESCRIPTION
**Pull request by nelsonni**
This update adds code to remove build directories and intermediateJSON.json files that might be present from previous compilation and build runs. Prior to this change, successive builds would append all new data to the previously built files and incorrectly duplicate the command log within the intermediateJSON.json file.

### CODE REVIEW

File: src/edu/oregonstate/edu/Replay.java [hunk 1]
```diff
@@ -77,10 +77,54 @@ public Boolean areFilesIdentical(String fileName1, String fileName2) {
...
+        }
+    }
+
     public void replayFile(String fileName) {
-//        if(replayDir.length()>0){
-//            fileName = replayDir + "/"+ fileName;
-//        }
+        // clean all files created on previous builds
+        System.out.println("Cleaning build directory...");
+        cleanBuildDir(replayDir, fileName);
```
Comment (cdmihai) [inline src/edu/oregonstate/edu/Replay.java]: What if you do this before replaying starts? 

That way you can just blindly delete the entire dir contents, which would eliminate a lot of logic in cleanBuildDir.

**Comment by cdmihai** [comment]
If you want, consider adding 3-4 tests for this functionality. Seems straightforward to test and it would eliminate the burden of manually verifying it.

**Comment by nelsonni** [comment]
I fixed and simplified the deletion process, as per suggestion. Tests will have to wait until tomorrow.

File: src/edu/oregonstate/edu/Replay.java [hunk 2]
```diff
@@ -208,6 +221,13 @@ private void addResource(JSONObject jObj) {
         }
         fileName = replayDir + "/" + fileName;
 
+        if (fileName.endsWith(".project")) {
+            String[] filePath = fileName.split("/");
+            String parentDir = fileName.substring(0, fileName.lastIndexOf('/'));
```
Comment (cdmihai) [inline src/edu/oregonstate/edu/Replay.java]: You can more clearly get the parent of a path by doing `Paths.get(filePath).getParent;`
The new java 1.7 Path API is quite awesome. Together with the Apache file utils it actually makes IO code quite pleasurable :)

**Pull request merged**